modules/standard/Sort.chpl

```diff
@@ -145,8 +210,20 @@ implemented with a compare method:
   // This will output: -1, 2, 3, -4
   writeln(Array);
 
+
+The keyPartComparator interface
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The keyPartComparator interface defines how a comparator should sort parts of
+a key using the ``keyPart`` method. This is used for certain sort algorithms.
+Records implementing this interface must define a ``keyPart`` method.
+
+A comparator implementing this interface can contain both ``compare`` and
+``keyPart`` methods. In that event, the sort algorithm will use whichever is
+appropriate for the algorithm and expect that they have consistent results.
```

Comment:
ws, "A comparator implementing this interface can optionally also provide a ``compare`` method."